services/dbaas/mariadb/cluster.go

```diff
@@ -284,9 +305,9 @@ func (c *Client) SetMariaDBClusterData(d *schema.ResourceData, cluster mariadb.C
 		}
 	}
 
-	if cluster.Properties.Connections != nil && len(*cluster.Properties.Connections) > 0 {
+	if len(cluster.Properties.Connections) > 0 {
```

Comment:
@rmocanu-ionos this is not included in the scope of this PR but I will make a note and open an issue about this:

Again, as we are doing for other resources, we are setting information about arrays only if the array is not empty, which is not correct, we need to always set information for the arrays, even if they are empty or not. In this case but also in others, if the array is emptied by an external source, Terraform won't see this change since the set is done only when the array is not empty.